### PR TITLE
ENH: Create semantic type for virsorter2 database

### LINKED ISSUE #1: ENH: Create semantic type for virsorter2 database

### PR DESCRIPTION
**Pull request by ChristosMatzoros**
This PR closes #1 .

- Defines a semantic type for the virsorter2 database.

The virsorter2 database is fetched once and then used multiple times for viromics analysis.

### Set up the environment
 ```shell
mamba create -n q2-viromics -c conda-forge -c bioconda -c  https://packages.qiime2.org/qiime2/2023.5/tested/ -c defaults q2-types numpy=1.23.5 q2cli virsorter=2 "python=3.8" scikit-learn=0.22.1 pandas seaborn hmmer==3.3 prodigal=2.6 screed=1 ruamel.yaml click pip last ncbi-genome-download checkv
```

###  Activate q2-viromics environment
```shell
conda activate q2-viromics
```

### Clone the repo and checkout the PR branch
```bash
git clone https://github.com/bokulich-lab/q2-viromics.git
cd q2-viromics
git fetch origin pull/2/head:pr-2
git checkout pr-2
pip install -e .
```

### 1. Download initial  virsorter2 database directory (9.8GB)
[here](https://polybox.ethz.ch/index.php/s/A5bKJ3aMHn5xNV0)

### Import database
```bash
# It may take some time!
qiime tools import --type 'Virsorter2Db' --input-path database --output-path vs2_db.qza
```
### Export database
```bash
qiime tools export --input-path vs2_db.qza --output-path vs2_db_exported
```
### Compare initial database directory with the exported directory from the artifact
```bash
git diff database vs2_db_exported
```

The expected content of the database directory is:
```bash
tree database
```

```
exported_db
├── Done_all_setup
├── group
│   ├── NCLDV
│   │   ├── hallmark-gene.list
│   │   ├── model
│   │   └── rbs-prodigal-train.db
│   ├── RNA
│   │   ├── hallmark-gene.list
│   │   └── model
│   ├── dsDNAphage
│   │   ├── hallmark-gene.list
│   │   └── model
│   ├── lavidaviridae
│   │   ├── hallmark-gene.list
│   │   └── model
│   └── ssDNA
│       ├── hallmark-gene.list
│       └── model
├── hmm
│   ├── pfam
│   │   ├── Pfam-A-Archaea.hmm
│   │   ├── Pfam-A-Bacteria.hmm
│   │   ├── Pfam-A-Eukaryota.hmm
│   │   ├── Pfam-A-Mixed.hmm
│   │   ├── Pfam-A-Viruses.hmm
│   │   └── Pfam-A-acc2desc.tsv
│   └── viral
│       └── combined.hmm
└── rbs
    ├── rbs-catetory-notes.tsv
    └── rbs-catetory.tsv

### CODE REVIEW

**Comment by Sann5** [comment]
Sorry, I haven't had time to review this @ChristosMatzoros. I'm gone thins Monday and maybe Tuesday so if it's urgent maybe you can ask someone else to do it. I will also prioritize finishing up stuff next week since it's my last one.

File: README.md [hunk 1]
```diff
@@ -25,13 +25,13 @@ cd q2-viromics
 Then, run:
 
 ```shell
-conda env create -n q2-viromics-dev --file ./environments/q2-viromics-qiime2-metagenome-2024.10.yml
+mamba create -n q2-viromics -c conda-forge -c bioconda -c  https://packages.qiime2.org/qiime2/2023.5/tested/ -c defaults q2-types numpy=1.23.5 q2cli virsorter=2 "python=3.8" scikit-learn=0.22.1 pandas seaborn hmmer==3.3 prodigal=2.6 screed=1 ruamel.yaml click pip last ncbi-genome-download checkv
```
Comment (VinzentRisch) [inline README.md]:
```suggestion
mamba create -n q2-viromics -c conda-forge -c bioconda -c  https://packages.qiime2.org/qiime2/2023.5/tested/ -c defaults q2-types numpy=1.23.5 q2cli virsorter=2 "python=3.8" scikit-learn=0.22.1 pandas seaborn hmmer==3.3 prodigal=2.6 screed=1 ruamel.yaml click pip last ncbi-genome-download checkv pyhmmer
```

File: setup.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 import versioneer
 
-description = ("A template QIIME 2 plugin.")
+description = "A QIIME 2 plugin for viromics identification and analysis."
```
Comment (VinzentRisch) [inline setup.py]:
```suggestion
description = "A QIIME 2 plugin for the identification and analysis of viral sequences."
```
identification of viromics does not make sense. Mine is a suggestion maybe you find something better.

**Review by VinzentRisch — CHANGES_REQUESTED** [review]
Looks very good. See some changes below.

**Comment by ChristosMatzoros** [comment]
Hey @VinzentRisch,

I did the corrections, you can have a look.

**Review by VinzentRisch — APPROVED** [review]
Looks good to me.

**Pull request merged**